app/code/Magento/Catalog/Model/Product/Copier.php

```diff
@@ -206,4 +229,29 @@ private function removeStockItem(array $productData): array
         }
         return $productData;
     }
+
+    /**
+     * Verify if generated url rewrite exists in url_rewrite table
+     *
+     * @param string $urlKey
+     * @return bool
+     */
+    private function isUrlRewriteExists(string $urlKey): bool
+    {
+        $urlRewriteCollection = $this->urlRewriteCollectionFactory->create();
+        $urlRewriteCollection->addFieldToFilter(UrlRewrite::ENTITY_TYPE, self::ENTITY_TYPE)
+            ->addFieldToFilter(UrlRewrite::REQUEST_PATH, $urlKey . $this->getUrlSuffix());
+
+        return $urlRewriteCollection->getSize() !== 0;
+    }
```

Comment:
We should not have a dependency on other module implementations, ideally, we should either depend on an API (interface) or handle this with a Plugin, in this case probably on the module `CatalogUrlRewrite`.